scipy/linalg/_decomp_update.pyx

```diff
@@ -0,0 +1,2416 @@
+            if typecode == cnp.NPY_FLOAT:
+                gemv(N, m, n, 1, <float*>qvoid, qs[0],
+                        <float*>uvoid, us[0], 0, col(<float*>qTuvoid, qTus, k), qTus[0])
+            if typecode == cnp.NPY_DOUBLE:
+                gemv(N, m, n, 1, <double*>qvoid, qs[0],
+                        <double*>uvoid, us[0], 0, col(<double*>qTuvoid, qTus, k), qTus[0])
+            if typecode == cnp.NPY_CFLOAT:
+                blas_t_conj(m, <float_complex*>uvoid, us)
+                gemv(N, m, n, 1, <float_complex*>qvoid, qs[0],
+                        <float_complex*>uvoid, us[0], 0,
+                        col(<float_complex*>qTuvoid, qTus, k), qTus[0])
+                blas_t_conj(m, col(<float_complex*>qTuvoid, qTus, k), qTus)
+            if typecode == cnp.NPY_CDOUBLE:
+                blas_t_conj(m, <double_complex*>uvoid, us)
+                gemv(N, m, n, 1, <double_complex*>qvoid, qs[0],
+                        <double_complex*>uvoid, us[0], 0,
+                        col(<double_complex*>qTuvoid, qTus, k), qTus[0])
+                blas_t_conj(m, col(<double_complex*>qTuvoid, qTus, k), qTus)
+        elif u.ndim == 2:
+            p = u.shape[1]
+            if cnp.PyArray_CHKFLAGS(u, cnp.NPY_F_CONTIGUOUS):
+                utrans = N
+                uvoid = extract(u, us)
+                ldu = us[1]
+            elif cnp.PyArray_CHKFLAGS(u, cnp.NPY_C_CONTIGUOUS):
+                utrans = T
+                uvoid = extract(u, us)
+                ldu = us[0]
+            else:
+                u = PyArray_FromArraySafe(u, NULL, cnp.NPY_F_CONTIGUOUS)
+                utrans = N
+                uvoid = extract(u, us)
+                ldu = us[1]
+            if typecode == cnp.NPY_FLOAT:
+                gemm(N, utrans, m, p, m, 1, <float*>qvoid, qs[0],
+                        <float*>uvoid, ldu, 0, col(<float*>qTuvoid, qTus, k), qTus[1])
+            elif typecode == cnp.NPY_DOUBLE:
+                gemm(N, utrans, m, p, m, 1, <double*>qvoid, qs[0],
+                        <double*>uvoid, ldu, 0, col(<double*>qTuvoid, qTus, k), qTus[1])
+            elif typecode == cnp.NPY_CFLOAT:
+                blas_t_2d_conj(m, p, <float_complex*>uvoid, us)
+                gemm(N, utrans, m, p, m, 1, <float_complex*>qvoid, qs[0],
+                        <float_complex*>uvoid, ldu, 0,
+                        col(<float_complex*>qTuvoid, qTus, k), qTus[1])
+                blas_t_2d_conj(m, p, col(<float_complex*>qTuvoid, qTus, k), qTus)
+            elif typecode == cnp.NPY_CDOUBLE:
+                blas_t_2d_conj(m, p, <double_complex*>uvoid, us)
+                gemm(N, utrans, m, p, m, 1, <double_complex*>qvoid, qs[0],
+                        <double_complex*>uvoid, ldu, 0,
+                        col(<double_complex*>qTuvoid, qTus, k), qTus[1])
+                blas_t_2d_conj(m, p, col(<double_complex*>qTuvoid, qTus, k), qTus)
+        else:
+            raise ValueError('1 <= u.ndim <= 2')
```

Comment:
Maybe expand this a bit for clarity: ``Shape of u is incorrect, should be 1 <= u.ndim <= 2``